### PR TITLE
Time variable

### PR DESCRIPTION
**Pull request by kernc**
RFC. Docs missing.

### CODE REVIEW

File: Orange/widgets/utils/__init__.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 def vartype(var):
     if var.is_discrete:
         return 1
-    elif var.is_continuous:
+    elif var.is_continuous and not isinstance(var, TimeVariable):
         return 2
     elif var.is_string:
         return 3
+    elif isinstance(var, TimeVariable):
+        return 4
```
Comment (lanzagar) [inline Orange/widgets/utils/__init__.py]: Not really important, but:

If we want this if-else flat and ordered, then I am in favour of changing it to go from 4 to 0.
TimeVariable (and future new types) can be added to the top and any unhandled variables are caught at the bottom as generic variable types.

Another option would be a nested if-else with all continuous types added inside the elif is_continuous. This shows the structure better, but loses the ordering and becomes more complex.

Comment (kernc) [inline Orange/widgets/utils/__init__.py]: This abomination of a mechanism shall not retain its current form for much longer. But mending it now is out of scope in this PR.

**Comment by lanzagar** [comment]
Looks good.
Will documentation be added as well?

**Comment by kernc** [comment]
Some docs have been added.

**Pull request merged**